bluepyefe/reader.py

```diff
@@ -166,153 +167,8 @@ def read_matlab(in_data):
-
-        if key_current not in content['stimulus']['presentation']:
-            continue
-
-        data.append(_format_nwb_trace(
-            voltage=content['acquisition'][sweep]['data'],
-            current=content['stimulus']['presentation'][key_current]['data'],
-            start_time=content["acquisition"][sweep]["starting_time"],
-            trace_name=sweep,
-        ))
-
-    return data
-
-
-def _get_repetition_keys_nwb(ecode_content, request_repetitions=None):
-
-    if isinstance(request_repetitions, (int, str)):
-        request_repetitions = [int(request_repetitions)]
-
-    reps = list(ecode_content.keys())
-    reps_id = [int(rep.replace("repetition ", "")) for rep in reps]
-
-    if request_repetitions:
-        return [reps[reps_id.index(i)] for i in request_repetitions]
-    else:
-        return list(ecode_content.keys())
-
-
-def _nwb_reader_BBP(content, target_protocols, repetition):
-
-    data = []
-
-    for ecode in target_protocols:
-
-        for cell_id in content["data_organization"].keys():
-
-            if ecode not in content["data_organization"][cell_id]:
-                logger.debug(f"No eCode {ecode} in nwb.")
-                continue
-
-            ecode_content = content["data_organization"][cell_id][ecode]
-
-            rep_iter = _get_repetition_keys_nwb(
-                ecode_content, request_repetitions=repetition
-            )
-
-            for rep in rep_iter:
-
-                for sweep in ecode_content[rep].keys():
-                    for trace_name in list(ecode_content[rep][sweep].keys()):
-
-                        if "ccs_" in trace_name:
-                            key_current = trace_name.replace("ccs_", "ccss_")
-                        elif "ic_" in trace_name:
-                            key_current = trace_name.replace("ic_", "ics_")
-                        else:
-                            continue
-
-                        if key_current not in content["stimulus"]["presentation"]:
-                            logger.debug(f"Ignoring {key_current} not"
-                                         " present in the stimulus presentation")
-                            continue
-                        if trace_name not in content["acquisition"]:
-                            logger.debug(f"Ignoring {trace_name} not"
-                                         " present in the acquisition")
-                            continue
-
-                        data.append(_format_nwb_trace(
-                            voltage=content["acquisition"][trace_name]["data"],
-                            current=content["stimulus"]["presentation"][key_current]["data"],
-                            start_time=content["stimulus"]["presentation"][key_current]["starting_time"],
-                            trace_name=trace_name,
-                            repetition=int(rep.replace("repetition ", ""))
-                        ))
-
-    return data
-
-
 def nwb_reader(in_data):
```

Comment:
Do we really need `in_data` to be a `dict`? It seems only the "filepath" is accessed by the function. What if we replace `in_data` with a string (or Path) that points to the file path NWB file?

E.g.
```python
def nwb_reader(filepath):
    ...
```

Reply:
We need it to be a dict as it can contain other metadata such as the v_unit, t_unit, ljp correction etc. Also it has to stay consistent between all readers.

Reply:
But that `in_data` does not go to the readers. Am I missing some detail?

Reply:
Oh I see what you meant. It's in cell.py.

Reply:
In that case how about the following?

Instead of this
```python
return nwb_reader(config_data)
```

This can be added
```python
return nwb_reader(config_data["filepath"])
```

So that nwb_reader takes the only parameter it needs (the filepath).

Reply:
Ooops it's a false positive. The other fields of in_data are used. The collapse view of github was obscuring it. All is well